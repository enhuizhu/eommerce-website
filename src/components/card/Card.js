import React from 'react';

export default (props) => {

  const product = props.product;

  return (
    <div className='card'>
      <div className='product-img'>
        <img 
          src={`/dist${product.image.path}`} 
          alt={product.image.alt}>
        </img>
      </div>

      <div className='label-container'>
        <label>{product.productLabel}</label>
      </div>

      <div className='product-detail'>
        <div className='product-title bold'>
          {product.title}
        </div>
        <div className='product-description'>
          {product.description}
        </div>
        <div className='product-price'>
          {product.priceLabel} 
          <label className='bold'>
            {`${product.currency}${product.price}`}
          </label>
        </div>

        <a href={product.ctaLink}>
          <div className='btn cta-btn bold'>
            {product.cta}
          </div>
        </a>
      </div>

    </div>
  );
}
