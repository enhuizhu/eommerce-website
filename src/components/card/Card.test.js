import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('test card component', () => {
  const product = {
    "title": "Simple Canvas",
    "description": "Lets your pictures speak for themselves.",
    "image": {
        "path": "/images/product.jpg",
        "alt": "Simple Canvas"
    },
    "price": 1500,
    "currency": "Â£",
    "priceLabel": "From",
    "productLabel": "bestseller",
    "cta": "Shop Now",
    "ctaLink": "/random/link/to/no/where"
  }
  
  it('renders correctly', () => {
    const tree = renderer
      .create(<Card product={product}></Card>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('testing content inside', () => {
    const card  = shallow(<Card product={product}></Card>);
    expect(card.contains(<img src='/dist/images/product.jpg' alt='Simple Canvas'></img>)).toBe(true);
})
});