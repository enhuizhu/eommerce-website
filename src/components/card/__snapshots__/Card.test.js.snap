// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card component renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="product-img"
  >
    <img
      alt="Simple Canvas"
      src="/dist/images/product.jpg"
    />
  </div>
  <div
    className="label-container"
  >
    <label>
      bestseller
    </label>
  </div>
  <div
    className="product-detail"
  >
    <div
      className="product-title bold"
    >
      Simple Canvas
    </div>
    <div
      className="product-description"
    >
      Lets your pictures speak for themselves.
    </div>
    <div
      className="product-price"
    >
      From
      <label
        className="bold"
      >
        Â£1500
      </label>
    </div>
    <a
      href="/random/link/to/no/where"
    >
      <div
        className="btn cta-btn bold"
      >
        Shop Now
      </div>
    </a>
  </div>
</div>
`;
